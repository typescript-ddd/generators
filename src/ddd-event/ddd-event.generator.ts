import { getTemplate } from "../shared/templates";
import { EventTemplateData } from "./ddd-event.types";

export const generateEvent = (data: EventTemplateData) => {
  const template = getEventTemplate();
  return template(data);
};

export const getEventTemplate = () =>
  getTemplate(
    __dirname + "/event.hbs"
  ) as HandlebarsTemplateDelegate<EventTemplateData>;
