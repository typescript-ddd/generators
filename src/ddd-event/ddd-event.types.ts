export interface EventTemplateData {
    entityName: string;
    eventType: string;
    eventTopic: string;
}