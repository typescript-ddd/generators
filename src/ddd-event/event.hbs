import { DomainEvent } from "@typescript-ddd/core";
import { {{pascal entityName}}, {{pascal entityName}}Id } from "../../domain/models";

/**
 * Represents {{noun entityName}} {{natural eventType}} event.
 */
export class {{pascal entityName}}{{pascal eventType}}Event extends DomainEvent<{{pascal entityName}}> {
  /**
   * Initializes a new instance of {{aOrAn entityName}} {{natural entityName}} {{natural eventType}} event.
   * @param {{lbrace}}{{pascal entityName}}Id{{rbrace}} id - The identifier of the {{natural entityName}}.
   * @param {{lbrace}}{{pascal entityName}}{{rbrace}} payload - The {{natural entityName}} payload.
   */
  constructor(id: {{pascal entityName}}Id, payload: {{pascal entityName}}) {
    super(id, "{{eventTopic}}", payload);
  }
}