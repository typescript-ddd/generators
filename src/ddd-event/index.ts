export * from "./ddd-event.generator";
export * from "./ddd-event.types";