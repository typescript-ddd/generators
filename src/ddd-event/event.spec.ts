import { generateEvent } from "./ddd-event.generator";
import { EventTemplateData } from "./ddd-event.types";

describe("Event", () => {
  it("should render a created event", () => {
    // Arrange
    const data: EventTemplateData = {
      entityName: "User",
      eventTopic: "user/created",
      eventType: "Created",
    };

    // Act
    const result = generateEvent(data);

    // Assert
    expect(result).toMatchSnapshot();
  });  
});
