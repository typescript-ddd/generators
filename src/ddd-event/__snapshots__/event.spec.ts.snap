// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event should render a created event 1`] = `
"import { DomainEvent } from "@typescript-ddd/core";
import { User, UserId } from "../../domain/models";

/**
 * Represents a user created event.
 */
export class UserCreatedEvent extends DomainEvent<User> {
  /**
   * Initializes a new instance of a user created event.
   * @param {UserId} id - The identifier of the user.
   * @param {User} payload - The user payload.
   */
  constructor(id: UserId, payload: User) {
    super(id, "user/created", payload);
  }
}"
`;
